src/InvestmentManager.sol

```diff
@@ -356,32 +384,33 @@ contract InvestmentManager is Auth {
         view
         returns (uint256 currencyAmount)
     {
-        uint128 depositPrice = calculateDepositPrice(user, liquidityPool);
+        uint256 depositPrice = calculateDepositPrice(user, liquidityPool);
         if (depositPrice == 0) return 0;
 
         currencyAmount = uint256(_calculateCurrencyAmount(_toUint128(trancheTokenAmount), liquidityPool, depositPrice));
     }
 
     /// @return trancheTokenAmount is type of uin256 to support the EIP4626 Liquidity Pool interface
-    function previewRedeem(address user, address liquidityPool, uint256 currencyAmount)
+    function previewWithdraw(address user, address liquidityPool, uint256 currencyAmount)
         public
         view
         returns (uint256 trancheTokenAmount)
     {
-        uint128 redeemPrice = calculateRedeemPrice(user, liquidityPool);
+        uint256 redeemPrice = calculateRedeemPrice(user, liquidityPool);
         if (redeemPrice == 0) return 0;
 
         trancheTokenAmount =
             uint256(_calculateTrancheTokenAmount(_toUint128(currencyAmount), liquidityPool, redeemPrice));
     }
 
     /// @return currencyAmount is type of uin256 to support the EIP4626 Liquidity Pool interface
-    function previewWithdraw(address user, address liquidityPool, uint256 trancheTokenAmount)
+    function previewRedeem(address user, address liquidityPool, uint256 trancheTokenAmount)
         public
         view
         returns (uint256 currencyAmount)
     {
-        uint128 redeemPrice = calculateRedeemPrice(user, liquidityPool);
+        require(trancheTokenAmount < type(uint128).max, "InvestmentManager/amount-too-large");
```

Comment:
Should we also add a similar check to `previewWithdraw()`?

Reply:
This was a leftover from some bugfixing, should have been removed.